grpcurl.go

```diff
@@ -609,7 +609,7 @@ func ServerTransportCredentials(cacertFile, serverCertFile, serverKeyFile string
 // BlockingDial is a helper method to dial the given address, using optional TLS credentials,
 // and blocking until the returned connection is ready. If the given credentials are nil, the
 // connection will be insecure (plain-text).
-func BlockingDial(ctx context.Context, network, address string, creds credentials.TransportCredentials, opts ...grpc.DialOption) (*grpc.ClientConn, error) {
+func BlockingDial(ctx context.Context, address string, creds credentials.TransportCredentials, opts ...grpc.DialOption) (*grpc.ClientConn, error) {
```

Comment:
This is a backwards-incompatible change. This is exported API and this module is post-v1.0. Changing the signature would require a v2.0, which is way more effort than is warranted for such a small change.

I think the following behavior would do what we need:
1. If network is the empty string, just use address as is.
2. If network is "tcp", return an error if address has a "unix://" prefix.
3. If network is "unix" and address does not have a "unix://" prefix, add one.
4. Otherwise, configure a custom dialer to actually dial using the selected network name.

So the command-line `main` package would always pass the empty string. But this way, we haven't broken any existing callers, including those that may be using other values (such as "tcp4" or "tcp6" to disable dual-stack support).